### PR TITLE
Log Watchdog events to scuba

### PR DESCRIPTION
**Pull request by bchen2020**
Summary: This diff logs some events of FileTimerServer to a scuba table. The events include "server started", "server stopped", "set timer", "clear timer" and "kill worker process".

Test Plan:
### Unit Test
```
buck test mode/dev-nosan //caffe2/test/distributed/elastic/agent/server/test:local_agent_test
```
```
Test Session: https://www.internalfb.com/intern/testinfra/testrun/1407375146936031
RE: reSessionID-2224cf79-6a28-4762-ab7c-9875adb244dc 3.4 KiB▲,  0.0 B▼
Jobs completed: 57. Time elapsed: 3084.4s.
Tests finished: Pass 55. Fail 0. Fatal 0. Skip 0. 0 builds failed
```

Differential Revision: D39665560

### CODE REVIEW

File: test/distributed/elastic/agent/server/test/local_elastic_agent_test.py [hunk 1]
```diff
@@ -32,8 +33,9 @@
 )
 from torch.distributed.elastic.agent.server.local_elastic_agent import (
     LocalElasticAgent,
-    TORCHELASTIC_TIMER_FILE,
+    TORCHELASTIC_ENABLE_FILE_TIMER,
 )
+from torch.distributed.elastic.events.fb.scuba import ScubaLogHandler
```
Comment (H-Huang) [inline test/distributed/elastic/agent/server/test/local_elastic_agent_test.py:38]: I'm surprised we don't see any failures from this. Are torch.elastic tests even running in our OSS CI? How can we be importing from internal fb-only code in our open source mirror? cc @d4l3k

**Review by H-Huang — CHANGES_REQUESTED** [review]
Curious about how scuba logging works with open source changes. Left a comment.

File: test/distributed/elastic/agent/server/test/local_elastic_agent_test.py (hunk 1 above)
Comment (d4l3k) [reply test/distributed/elastic/agent/server/test/local_elastic_agent_test.py:38]: @H-Huang I'm not 100% sure -- how are are the builds configured in OSS CI? / how do I check? 

@kiukchung would probably have the best context here

Comment (bchen2020) [reply test/distributed/elastic/agent/server/test/local_elastic_agent_test.py]: I moved the scuba test to a FB internal test module.

**Comment by bchen2020** [comment]
> 



> Curious about how scuba logging works with open source changes. Left a comment.

The _log_handlers in open source maps the "null" destination to a logging.NullHandler, while with FB internal code, it maps to a scuba log handler.

**Review by d4l3k — APPROVED** [review]
would be nice to have logging test in OSS

File: test/distributed/elastic/agent/server/test/local_elastic_agent_test.py (hunk 1 above)
Comment (kiukchung) [reply test/distributed/elastic/agent/server/test/local_elastic_agent_test.py]: the only tests we "blocklist" are defined [here](https://github.com/pytorch/pytorch/blob/master/test/run_test.py#L95), which doesn't list `distributed.elastic.agent.server.test.local_elastic_agent_test`. Did the CI that was triggered for this PR not run `local_elastic_agent_test.py`?

**Pull request closed without merging**